server/repository/src/db_diesel/document.rs

```diff
@@ -394,37 +411,38 @@ impl DocumentRow {
             data,
             form_schema_id,
             status,
-            owner_name_id,
+            owner_name_link_id: _,
             context_id,
-        } = self;
+        },
+        owner_name_join,
+    ) = join;
 
-        let parents: Vec<String> =
-            serde_json::from_str(&parent_ids).map_err(|err| RepositoryError::DBError {
-                msg: "Invalid parents data".to_string(),
-                extra: format!("{}", err),
-            })?;
-        let data: serde_json::Value =
-            serde_json::from_str(&data).map_err(|err| RepositoryError::DBError {
-                msg: "Invalid data".to_string(),
-                extra: format!("{}", err),
-            })?;
+    let parents: Vec<String> =
+        serde_json::from_str(&parent_ids).map_err(|err| RepositoryError::DBError {
+            msg: "Invalid parents data".to_string(),
+            extra: format!("{}", err),
+        })?;
+    let data: serde_json::Value =
+        serde_json::from_str(&data).map_err(|err| RepositoryError::DBError {
+            msg: "Invalid data".to_string(),
+            extra: format!("{}", err),
+        })?;
 
-        let document = Document {
-            id,
-            name,
-            parent_ids: parents,
-            user_id,
-            datetime: DateTime::<Utc>::from_utc(datetime, Utc),
-            r#type,
-            data,
-            form_schema_id,
-            status,
-            owner_name_id,
-            context_id,
-        };
+    let document = Document {
+        id,
+        name,
+        parent_ids: parents,
+        user_id,
+        datetime: DateTime::<Utc>::from_utc(datetime, Utc),
+        r#type,
+        data,
+        form_schema_id,
+        status,
+        owner_name_id: owner_name_join.map(|(_, name_row)| name_row.id),
```

Comment:
Should this be `owner_name_link_id` since you've done it for the `DocumentRow`?

Reply:
Document has the correct naming, so I really want the `owner_name_id` (there is an issue for fixing it for other repos as well...) I already have done it here because it was easy to do because I have to map the DocumentRow to a Document anyway...